src/components/Editor/ClientCursor.ts

```diff
@@ -29,57 +31,69 @@ class ClientCursor {
-    cursorElement.style.borderLeftWidth = '2px';
+    cursorElement.style.borderLeftWidth = '2.5px';
     cursorElement.style.borderLeftColor = this.color;
-    cursorElement.style.height = `${size}px`;
+    cursorElement.style.height = `${this.height}px`;
     cursorElement.style.padding = '0px';
-    cursorElement.style.zIndex = '0';
 
-    const nameBoxElement = document.createElement('span');
-    nameBoxElement.style.position = 'absolute';
-    nameBoxElement.style.top = `-${size}px`;
-
-    const nameElement = document.createElement('span');
-    nameElement.textContent = this.id;
-    nameElement.style.position = 'fixed';
-    nameElement.style.backgroundColor = this.color;
-    nameElement.style.padding = '1px 4px';
-    nameElement.style.borderRadius = '4px';
-    nameElement.style.color = invert(this.color, true);
-    nameElement.style.animationDuration = `${duration}s`;
-    nameElement.style.animationDelay = `${delay}s`;
-    nameElement.className = 'text-remove';
-
-    nameBoxElement.appendChild(nameElement);
-    cursorElement.appendChild(nameBoxElement);
-
-    this.removeNameReserve(nameBoxElement);
+    this.showCursorNameReserve(cursorElement);
     this.marker = cm.setBookmark(cursorPos, {
       widget: cursorElement,
       insertLeft: true,
     });
   }
 
-  updateLine(cm: CodeMirror.Editor, fromPos: CodeMirror.Position, toPos: CodeMirror.Position) {
+  updateLine(
+    cm: CodeMirror.Editor,
+    fromPos: CodeMirror.Position,
+    toPos: CodeMirror.Position,
+  ) {
     this.removeLine();
     this.lineMarker = cm.getDoc().markText(fromPos, toPos, {
       css: `background-color : ${this.color}; opacity:0.7`,
     });
   }
 
+  // when user's cursor hover, show name
+  showCursorNameReserve(parentEl: Element) {
+    const nameElement = document.createElement('span');
+
+    parentEl.addEventListener('mouseenter', () => {
+      nameElement.textContent = this.id;
+      nameElement.style.position = 'absolute';
+      nameElement.style.top = `-${this.height}px`;
+      nameElement.style.left = `-2px`;
+      nameElement.style.backgroundColor = this.color;
+      nameElement.style.padding = '1px 4px';
+      nameElement.style.borderRadius = '4px 4px 4px 0px';
+      nameElement.style.color = invert(this.color, true);
+      nameElement.style.animationDuration = `${duration}s`;
+      nameElement.style.animationDelay = `${delay}s`;
+      nameElement.className = 'text-remove';
+
+      parentEl.appendChild(nameElement);
+    });
+
+    parentEl.addEventListener('mouseleave', () => {
+      this.removeNameReserve(nameElement);
+    });
+  }
+
   // After animate, It should actually be deleted it.
   removeNameReserve(el: Element) {
     if (markerRemoveMap.has(this.id)) {
       clearTimeout(markerRemoveMap.get(this.id));
     }
 
     const timeoutId = window.setTimeout(() => {
-      el.parentNode!.removeChild(el);
-      markerRemoveMap.delete(this.id);
+      if (el.parentNode) {
+        el.parentNode!.removeChild(el);
```

Comment:
Is the `!` required inside conditional statements?

Reply:
I think I've seen this case in other code

Come to think of it, it would be better to remove. because it was considered a duplicate!